modules/hashi-snapshots.nix

```
@@ -0,0 +1,416 @@
+{
+  config,
+  pkgs,
+  lib,
+  hashiTokens,
+  ...
+}: let
+  cfg = config.services.hashi-snapshots;
+
+  inherit (lib) boolToString listToAttrs mkEnableOption mkIf mkMerge mkOption nameValuePair toUpper;
+  inherit (lib.types) addCheck attrs bool enum int str submodule;
+
+  snapshotJobConfig = submodule {
+    options = {
+      enable = mkOption {
+        type = bool;
+        default = true;
+        description = ''
+          Creates a systemd service and timer to automatically save Vault snapshots.
+        '';
+      };
+
+      backupCount = mkOption {
+        type = addCheck int (x: x >= 0);
```

```suggestion
        type = ints.unsigned;
```

(and `inherit (lib.types) … ints;` above)

Same in line 117.